modules/rostests/kmtests/rtl/RtlStack.c

```diff
@@ -130,7 +130,71 @@ TestStackWalk1(VOID)
     TestStackWalk2();
 }
 
+#ifdef _M_AMD64
+NTSYSAPI
+PVOID
+NTAPI
+RtlPcToFileHeader(
+    _In_  PVOID PcValue,
+    _Out_ PVOID *BaseOfImage);
+
+extern char __ImageBase;
+
+DECLSPEC_NOINLINE
+static
+VOID
+TestRtlPcToFileHeader(VOID)
+{
+    PVOID ImageBase, Result;
+    PTEB Teb;
+    PPEB Peb;
+
+    /* First test a function from this image */
+    Result = RtlPcToFileHeader(&TestRtlPcToFileHeader, &ImageBase);
+    ok_eq_pointer(Result, ImageBase);
+    ok_eq_pointer(ImageBase, &__ImageBase);
+
+#ifdef NTOS_MODE_USER
+    Teb = NtCurrentTeb();
+#else
+    Teb = KeGetCurrentThread()->Teb;
+#endif
+    ok(Teb != NULL, "Teb is NULL!");
+    if (Teb == NULL)
+    {
+        return;
+    }
+
+    _SEH2_TRY
+    {
+        Peb = Teb->ProcessEnvironmentBlock;
+        ok(Peb != NULL, "Peb is NULL!");
+        if (Peb == NULL)
+        {
+            return;
+        }
+
+        /* Test an address somewhere within the main image of the current process */
```

Comment:
Is there any point in testing how the UM function behaves with a kernel-mode address?

Reply:
I don't think so, but feel free to add it.